### ISSUE TITLE
Fatal error when accessing componentless translation view

### ISSUE THREAD

**Issue by rageandengage** [WeblateOrg/weblate#4615]
**Describe the bug**
If a user can view a project, but they can not view any components, and they navigate to https://URL/translate/project-slug/-/lang-code/, their request will throw an exception.

In our case:
```
Request Method: GET
Request URL: https://project.url/translate/PROJECT/-/zh_Hans/
Django Version: 3.1.1
Exception Type: ValueError
Exception Value: Not supported type for permission check: ProjectLanguageComponent
Exception Location: weblate/auth/permissions.py, line 84, in check_permission
```

This occurs because the default component class in weblate/utils/stats.py is `self.component = ProjectLanguageComponent()`

**Comment by nijel** [comment]
Should be already fixed by 9512fbe3ac8ec8a5bc81df02e08188c62661f655, in case it doesn't fix it please provide full exception traceback and Weblate version information.

**Comment by rageandengage** [comment]
> Should be already fixed by [9512fbe](https://github.com/WeblateOrg/weblate/commit/9512fbe3ac8ec8a5bc81df02e08188c62661f655), in case it doesn't fix it please provide full exception traceback and Weblate version information.

Looks like 9512fbe resolves the issue. We're running on latest stable (4.2.2) so I didn't catch it. Thank you
